core/dataset.cpp

```diff
@@ -232,72 +232,46 @@ void Dataset::setData(const std::string &name, Variable data) {
   m_data[name].data = std::move(data);
 }
 
-// This only checks the coordinates and labels, not attributes
-bool checkCorrespondingDenseCoords(const Dataset &dataset,
-                                   const DataConstProxy &other) {
-  if (other.dims().sparse())
-    return true;
-  const auto dsCoords{dataset.coords()};
-  const auto otCoords{other.coords()};
-  const auto &dsItems = dsCoords.items();
-  for (const auto & [ d, v ] : otCoords) {
-    if (auto iter = dsItems.find(d); iter == dsItems.end()) {
-      return false;
+/// Set (insert or replace) data item with given name.
+///
+/// Coordinated, labels, and attributes of the data array are added to the
```

Comment:
Can't see if attributes are added.

Reply:
Good point, they are not, since that would require merging rules, which we have not defined yet (and may never). I will fix the doxygen, it is weong.